webapi/src/main/scala/org/knora/webapi/routing/HealthRouteWithZIOHttp.scala

```diff
@@ -0,0 +1,117 @@
+ */
+
+package org.knora.webapi.routing
+
+import spray.json.JsObject
+import spray.json.JsString
+import zhttp.http._
+import zio._
+
+import org.knora.webapi.core.State
+import org.knora.webapi.core.domain.AppState
+
+/**
+ * Provides health check logic
+ */
+trait HealthCheckWithZIOHttp {
+
+  protected def healthCheck(state: State): UIO[Response] =
+    for {
+      _        <- ZIO.logDebug("get application state")
+      state    <- state.getAppState
+      result   <- setHealthState(state)
+      _        <- ZIO.logDebug("set health state")
+      response <- createResponse(result)
+      _        <- ZIO.logDebug("getting application state done")
+    } yield response
+
+  private def setHealthState(state: AppState): UIO[HealthCheckResult] =
+    ZIO.succeed(
+      state match {
+        case AppState.Stopped                => unhealthy("Stopped. Please retry later.")
+        case AppState.StartingUp             => unhealthy("Starting up. Please retry later.")
+        case AppState.WaitingForTriplestore  => unhealthy("Waiting for triplestore. Please retry later.")
+        case AppState.TriplestoreReady       => unhealthy("Triplestore ready. Please retry later.")
+        case AppState.UpdatingRepository     => unhealthy("Updating repository. Please retry later.")
+        case AppState.RepositoryUpToDate     => unhealthy("Repository up to date. Please retry later.")
+        case AppState.CreatingCaches         => unhealthy("Creating caches. Please retry later.")
+        case AppState.CachesReady            => unhealthy("Caches ready. Please retry later.")
+        case AppState.UpdatingSearchIndex    => unhealthy("Updating search index. Please retry later.")
+        case AppState.SearchIndexReady       => unhealthy("Search index ready. Please retry later.")
+        case AppState.LoadingOntologies      => unhealthy("Loading ontologies. Please retry later.")
+        case AppState.OntologiesReady        => unhealthy("Ontologies ready. Please retry later.")
+        case AppState.WaitingForIIIFService  => unhealthy("Waiting for IIIF service. Please retry later.")
+        case AppState.IIIFServiceReady       => unhealthy("IIIF service ready. Please retry later.")
+        case AppState.WaitingForCacheService => unhealthy("Waiting for cache service. Please retry later.")
+        case AppState.CacheServiceReady      => unhealthy("Cache service ready. Please retry later.")
+        case AppState.MaintenanceMode        => unhealthy("Application is in maintenance mode. Please retry later.")
+        case AppState.Running                => healthy
+      }
+    )
+
+  private def createResponse(result: HealthCheckResult): UIO[Response] =
+    ZIO.succeed(
+      Response
+        .json(
+          JsObject(
+            "name"     -> JsString("AppState"),
+            "severity" -> JsString("non fatal"),
+            "status"   -> JsString(status(result.status)),
+            "message"  -> JsString(result.message)
+          ).toString()
```

Comment:
could we use ZioJson here instead?

Reply:
also, using a hard-coded sting for name and severity, when we actually have the identical string in the object seems very redundant

Reply:
I thought the same but I only touched the code in order to get it to compile - nothing more. I don't know why we need the "AppState" and "non fatal" if it is hard-coded and in both cases (healthy/unhealthy) the same anyway. Maybe @subotic  has an opinion about it?

Reply:
I think generally it's more important to replace sprayJson with zioJson than getting rid of two hardcoded strings

Reply:
I see and you are totally right. There is already an epic for refactoring the health route: https://dasch.atlassian.net/browse/DEV-1218, so I classify your input to "out of scope" here and add it to that issue.